src/test/java/org/opensearch/neuralsearch/common/BaseNeuralSearchIT.java

```diff
@@ -68,15 +76,33 @@ public abstract class BaseNeuralSearchIT extends OpenSearchSecureRestTestCase {
 
     protected final ClassLoader classLoader = this.getClass().getClassLoader();
 
+    protected ThreadPool threadPool;
+    protected ClusterService clusterService;
+
     protected void setPipelineConfigurationName(String pipelineConfigurationName) {
         this.PIPELINE_CONFIGURATION_NAME = pipelineConfigurationName;
     }
 
     @Before
     public void setupSettings() {
+        threadPool = setUpThreadPool();
+        clusterService = createClusterService(threadPool);
```

Comment:
do we really need this? what are really testing with these, as these are integration tests

Reply:
Yes we need this because when we get node version from cluster. this.clusterService.getState.getNode.getMinNodeVersion.

While executing integ test case of NeuralQueryProcessor, **this.clusterService** throws null pointer if we do not mock it.

Reply:
But these are integration steps, why do I need to mock it?

Reply:
Oops I used the wrong word here, instead of mock I should I have used initialize. Opensearch has created a test suite for integ test called OpenSearchIntegTestCase. We use is BaseNeuralSearchIT in our test cases. Now in OpenSearchIntegTestCase they [initialize](https://github.com/opensearch-project/OpenSearch/blob/8f4f9951de817a58d18815c5dd35896dae2bced1/test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java#L654) clusterService. In our case we don't initalize so that is the reason I was getting null in NeuralSearchClusterUtil. Now I have initialized.